aging_due_report/wizard/wizard.py

```diff
@@ -585,9 +511,61 @@ def compute_lines(self, partner_ids):
                 aawp_ids[partner_id, currency_id] = aawp_id.id
             line['aawp_id'] = aawp_ids[partner_id, currency_id]
             line['aaw_id'] = wzd_brw.id
+            line['due_days'] = self._get_due_days(today, line)
             aawd_obj.create(line)
+
+        # Only to trigger _get_amount
+        wzd_brw._get_amount()
         return True
 
+    def _get_amount(self):
+        field_sum = ['residual', 'payment', 'total', 'not_due']
+        field_spans = ['span01', 'span02', 'span03', 'span04', 'span05']
+        field_names = field_spans + field_sum
+        direction = self.direction == 'past'
+        spans = [
+            self.period_length * x * (direction and 1 or -1)
+            for x in range(5)]
+
+        for wzd_cur in self.currency_ids:
+            residual = not_due = span01 = span02 = span03 = span04 = 0.0
+            span05 = 0.0
+            for part in wzd_cur.partner_ids:
+                res = {}.fromkeys(field_names, 0.0)
+                for doc in part.document_ids:
+                    for fsum in field_sum:
+                        if fsum != 'not_due':
+                            res[fsum] += doc.read([fsum])[0][fsum]
```

Comment:
We have too many small queries here yet.

What about use the following patch with `read_group`?
https://gist.github.com/moylop260/9e0211ee02cd440101d733f0e7f337c3#file-read_group-patch